src/jit/MemoryInl.h

```diff
@@ -1543,4 +1543,118 @@ static void emitAtomic(sljit_compiler* compiler, Instruction* instr)
 #undef OP_XCHG
 #undef OP_CMPXCHG
 
+static sljit_s32 atomicWaitCallback(ExecutionContext* context, sljit_s32 memOffset, sljit_s32 size)
+{
+    Instance* instance = context->instance;
+    if (!instance->memory(0)->isShared()) {
+        return ExecutionContext::ExpectedSharedMemError;
+    }
+    uint32_t result = 0;
+    int64_t timeout = context->tmp2[0];
+    int64_t expect = context->tmp1[0];
+    if (size == 8) {
+        instance->memory(0)->atomicWait(context->state, instance->module()->store(), memOffset, 0, expect, timeout, &result);
+    } else {
+        instance->memory(0)->atomicWait(context->state, instance->module()->store(), memOffset, 0, (int32_t)expect, timeout, &result);
+    }
+    context->tmp2[0] = result;
+    return ExecutionContext::NoError;
+}
+
+static void emitAtomicWait(sljit_compiler* compiler, Instruction* instr)
+{
+    CompileContext* context = CompileContext::get(compiler);
+    sljit_s32 size = (instr->opcode() == ByteCode::MemoryAtomicWait64Opcode ? 8 : 4);
+
+    MemoryAtomicWait32* atomicWait32Operation = reinterpret_cast<MemoryAtomicWait32*>(instr->byteCode());
```

Comment:
Is this type casting is valid for AtomicWait64?

Reply:
Still valid.

Reply:
This looks somewhat confusing, because `MemoryAtomicWait32` and `MemoryAtomicWait64` are handled together here.
Would it be better to merge these two bytecodes into one unfied bytecode like `MemoryAtomicWait`?